flag-alt-text.sh

```diff
@@ -1,13 +1,16 @@
 #!/bin/bash
 
 flagAltText() {
-  markdownMacOsScreenshotRegex="^.*!\[(Clean|Screen) ?[S|s]hot [0-9][0-9][0-9][0-9]-[0-1][0-9]-[0-9][0-9].*\].*$"
-  semanticMacOsScreenshotRegex="^.*<img.*(Clean|Screen) ?[S|s]hot [0-9][0-9][0-9][0-9]-[0-1][0-9]-[0-9][0-9].*$"
-  markdownImageRegex="^.*!\[(i|I)mage\].*$"
-  semanticImageRegex="^.*<img.*alt=\"(i|I)mage\".*$"
-  if  [[ $1 =~ $semanticMacOsScreenshotRegex ]] || [[ $1 =~ $markdownMacOsScreenshotRegex ]] || [[ $1 =~ $semanticImageRegex ]] || [[ $1 =~ $markdownImageRegex ]]; then
-    echo true
-  else
-    echo false
-  fi
+    markdownMacOsScreenshotRegex="^.*!\[(Clean|Screen) ?[S|s]hot [0-9][0-9][0-9][0-9]-[0-1][0-9]-[0-9][0-9].*\].*$"
+    semanticMacOsScreenshotRegex="^.*<img.*(Clean|Screen) ?[S|s]hot [0-9][0-9][0-9][0-9]-[0-1][0-9]-[0-9][0-9].*$"
+    markdownImageRegex="^.*!\[(i|I)mage\].*$"
+    semanticImageRegex="^.*<img.*alt=\"(i|I)mage\".*$"
+    emptySemanticRegex="^.*<img.*alt=(\"|')(\"|').*$"
+    emptyMarkdownRegex="^.*!\[\].*$"
+
+    if [[ $1 =~ $semanticMacOsScreenshotRegex ]] || [[ $1 =~ $markdownMacOsScreenshotRegex ]] || [[ $1 =~ $semanticImageRegex ]] || [[ $1 =~ $markdownImageRegex ]] || [[ $1 =~ $emptySemanticRegex ]] || [[ $1 =~ $emptyMarkdownRegex ]]; then
```

Comment:
Could we possibly check if something IS an HTML img, but doesn't have `alt=` using the NOT operator?

Syntax might be off, but something like:
```
    semanticImageRegex="^.*<img.*$"
    semanticImageWithAltRegex="^.*<img.*alt=.*$"
```
then in the if conditional we have:

```
|| [[ ($1 =~ $semanticImageRegex) && !($1 =~ $semanticImageWithAltRegex) ]]
```

Reply:
Love this idea let me throw it in